### PR TITLE
update the sandbox code to work with the latest stable bundles

### PR DESCRIPTION
**Pull request by rande**
Make the sandbox work again ...

This PR need to be merged first and new tags created:
- [x] https://github.com/sonata-project/SonataPageBundle/pull/708
- [x] https://github.com/sonata-project/SonataNewsBundle/pull/337
- [x] https://github.com/sonata-project/ecommerce/pull/387
- [x] https://github.com/sonata-project/SonataAdminBundle/pull/4083
- [x] https://github.com/sonata-project/SonataAdminBundle/pull/4081

Target the 2.4-develop branch as it is the next version.

Should fix many issues:
- https://github.com/sonata-project/sandbox/issues/580
- https://github.com/sonata-project/sandbox/issues/576
- https://github.com/sonata-project/sandbox/issues/572
- And many more ;)

### CODE REVIEW

**Comment by core23** [comment]
Can we add the composer key to sort the dependencies?

**Comment by rande** [comment]
@core23 can you show me a hint ?

**Comment by greg0ire** [comment]
Here is a hint : https://github.com/sonata-project/SonataAdminBundle/blob/3.x/composer.json#L73

File: src/Sonata/Bundle/QABundle/Tests/SonataAdminExplainCommandTest.php [hunk 1]
```diff
@@ -31,6 +31,6 @@ public function testExplain($id, $class)
 
         $output = $this->runCommand($client, sprintf("sonata:admin:explain %s", $id));
 
-        $this->assertNotNull($output);
+        $this->assertNotNull($output, sprintf("Fail to assert admin id: %s with class: %s", $id, $class));
     }
 }
```
Comment (OskarStark) [inline src/Sonata/Bundle/QABundle/Tests/SonataAdminExplainCommandTest.php]: ping

File: app/config/routing_front.yml [hunk 2]
```diff
@@ -78,6 +78,6 @@ blog:
 
 fos_comment:
     type: rest
-    resource: "@FOSCommentBundle/Resources/config/routing.yml"
+    resource: '@FOSCommentBundle/Resources/config/routing.yml'
     prefix: /comments
     defaults: { _format: html }
```
Comment (OskarStark) [inline app/config/routing_front.yml]: ping

File: src/Sonata/Bundle/QABundle/Tests/SonataAdminExplainCommandTest.php (hunk 1 above)
Comment (rande) [inline src/Sonata/Bundle/QABundle/Tests/SonataAdminExplainCommandTest.php]: ?

Comment (OskarStark) [inline src/Sonata/Bundle/QABundle/Tests/SonataAdminExplainCommandTest.php]: sorry, new line ending missing 😄

File: app/config/routing_front.yml (hunk 2 above)
Comment (OskarStark) [inline app/config/routing_front.yml]: new line ending missing 😄

**Comment by rande** [comment]
@OskarStark I just run php-cs-fixer ;) should be better now

**Comment by core23** [comment]
Sure we should use the `2.4-develop` branch for this? Wouldn't `3.0` make more sense, because most dependencies have a `3.x`. release.

**Comment by rande** [comment]
@core23 the naming is wrong, once merged we can clean branch names.

**Comment by core23** [comment]
So the next release on this branch will be a 3.0 release?

File: Makefile [hunk 3]
```diff
@@ -64,6 +64,12 @@ assets:
 assets-watch:
 	app/console assetic:dump --watch
 
+format:
+	php-cs-fixer fix src
+	rm -rf app/cache/*
+	php-cs-fixer fix app
+
```
Comment (OskarStark) [inline Makefile]: 1 blank line is enough ;-)

**Comment by core23** [comment]
@rande Can you finish this or are there any open issues?

**Comment by core23** [comment]
Can we update the sandbox @rande ? Anything to do here?

**Comment by core23** [comment]
Should we merge this and fix the open issues later? @sonata-project/contributors 

Looks like @rande didn't have time for this ATM

**Comment by greg0ire** [comment]
Ok for me

**Comment by core23** [comment]
@OskarStark @Soullivaneuh Feel free to merge :)

**Comment by OskarStark** [comment]
Thank you @rande 👍

**Comment by davask** [comment]
@core23 as you deleted the fix_installation branch which one am I suppose to use in replacement ?

**Comment by greg0ire** [comment]
yes

**Pull request merged**